apiv6/kentikapi/httputil/httputil.go

```diff
@@ -39,8 +37,141 @@ func NewRetryingClient(cfg ClientConfig) *http.Client {
 	if cfg.RetryWaitMax != nil {
 		c.RetryWaitMax = *cfg.RetryWaitMax
 	}
-	c.CheckRetry = retryablehttp.ErrorPropagatedRetryPolicy
+	c.CheckRetry = makeRetryPolicy(cfg.RetryableStatusCodes, cfg.RetryableMethods)
 	c.ErrorHandler = retryablehttp.PassthroughErrorHandler
 
 	return c.StandardClient()
 }
+
+// ClientConfig holds configuration for retrying client.
+type ClientConfig struct {
+	HTTPClient           *http.Client
+	RetryMax             *int
```

Comment:
The Config and ClientConfig retry-related attributes could be extracted into dedicated struct like HTTPRetryConfig as they repeat and make pretty cohesive, single-purpose pack that could be documented in single place